lightning/src/onion_message/messenger.rs

```diff
@@ -362,75 +362,71 @@ impl Responder {
 		}
 	}
 
-	/// Creates a [`ResponseInstruction::WithoutReplyPath`] for a given response.
+	/// Creates a [`ResponseInstruction`] for responding without including a reply path.
 	///
 	/// Use when the recipient doesn't need to send back a reply to us.
-	pub fn respond<T: OnionMessageContents>(self, response: T) -> ResponseInstruction<T> {
-		ResponseInstruction::WithoutReplyPath(OnionMessageResponse {
-			message: response,
-			reply_path: self.reply_path,
-		})
+	pub fn respond(self) -> ResponseInstruction {
+		ResponseInstruction {
+			destination: Destination::BlindedPath(self.reply_path),
+			context: None,
+		}
 	}
 
-	/// Creates a [`ResponseInstruction::WithReplyPath`] for a given response.
+	/// Creates a [`ResponseInstruction`] for responding including a reply path.
 	///
 	/// Use when the recipient needs to send back a reply to us.
-	pub fn respond_with_reply_path<T: OnionMessageContents>(self, response: T, context: MessageContext) -> ResponseInstruction<T> {
-		ResponseInstruction::WithReplyPath(OnionMessageResponse {
-			message: response,
-			reply_path: self.reply_path,
-		}, context)
+	pub fn respond_with_reply_path(self, context: MessageContext) -> ResponseInstruction {
+		ResponseInstruction {
+			destination: Destination::BlindedPath(self.reply_path),
+			context: Some(context),
+		}
 	}
 }
 
-/// This struct contains the information needed to reply to a received message.
-pub struct OnionMessageResponse<T: OnionMessageContents> {
-	message: T,
-	reply_path: BlindedMessagePath,
-}
-
-/// `ResponseInstruction` represents instructions for responding to received messages.
-pub enum ResponseInstruction<T: OnionMessageContents> {
-	/// Indicates that a response should be sent including a reply path for
-	/// the recipient to respond back.
-	WithReplyPath(OnionMessageResponse<T>, MessageContext),
-	/// Indicates that a response should be sent without including a reply path
-	/// for the recipient to respond back.
-	WithoutReplyPath(OnionMessageResponse<T>),
-	/// Indicates that there's no response to send back.
-	NoResponse,
+/// Instructions for how and where to send the response to an onion message.
+#[derive(Clone)]
+pub struct ResponseInstruction {
+	destination: Destination,
```

Comment:
Is there any reason to make this a `Destination` instead of just a `BlindedMessagePath`? Sorry if I missed a discussion here, I see this https://github.com/lightningdevkit/rust-lightning/pull/3263#discussion_r1725949731 but Jeff seems to have the same suggestion. It seems like any response is by definition sent over a reply *path*...

Reply:
Ah, okay.